crates/module-system/sov-modules-api/src/containers/vec.rs

```diff
@@ -320,6 +320,36 @@ where
     }
 }
 
+impl<V, Codec> NamespacedStateVec<Accessory, V, Codec>
+where
+    Codec: StateCodec,
+    Codec::ValueCodec: StateItemCodec<V> + StateItemCodec<u64>,
+    Codec::KeyCodec: StateItemCodec<u64>,
+{
+    /// Pushes a value to the end of the vector without requiring read access to the length.
+    pub fn push_accessory<Vq, W>(
+        &mut self,
+        value: &Vq,
+        state: &mut W,
+    ) -> Result<(), <W as StateWriter<Accessory>>::Error>
+    where
+        W: StateWriter<Accessory>,
+        Vq: ?Sized,
+        Codec::ValueCodec: EncodeLike<Vq, V>,
+    {
+        let len_key = self.len_value.slot_key();
+        let index = if let Some(len_bytes) = state.get_value(Accessory::NAMESPACE, &len_key) {
+            self.len_value
+                .codec()
+                .value_codec()
+                .decode_unwrap(len_bytes.value())
+        } else {
+            0
+        };
+        self.len_value.set(&index.saturating_add(1), state)?;
```

Comment:
Shouldn’t we return an error instead of using `index.saturating_add(1)` if the length overflows?

Reply:
Hmm, I think you're right that failing silently is a bad idea. This *should* be unreachable since we use a u64 for the index, but it seems much better to panic if that assumption is violated than to have silent data corruption. Fixed this and a few other instances!